### PR TITLE
[PLAYER-1934] Chapter Markers 

### PR DESCRIPTION
**Pull request by marcosmx**
This branch contains the following new components:

- marker: This represents a region on the scrubber bar to show whenever a marker starts and ends.
- markerIcon: This will render an icon with text or an image that could be expanded in order to show more information. This component will allow the user to perform a seek action when a click event it's detected on it.

### CODE REVIEW

File: tests/components/markers/marker-test.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+import React from 'react';
+import Enzyme from 'enzyme';
+import Marker from '../../../js/components/markers/marker';
+
+describe('Marker component', function () {
+
+  let baseProps;
+
+  function renderComponent() {
```
Comment (ooyala-amohnatkin) [inline tests/components/markers/marker-test.js]: Please don't use a mutable object from closure, pass props directly to this function. Otherwise, it's hard to follow what's going on

Comment (marcosmx) [reply tests/components/markers/marker-test.js]: Thanks, good catch!

Comment (marcosmx) [reply tests/components/markers/marker-test.js]: Done. I made a few changes to pass a new set of props to that function.

File: tests/components/markers/markerIcon-test.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    baseProps.data.text = 'test';
+    baseProps.accentColor = '#0000';
+    baseProps.duration = 300;
+    baseProps.scrubberBarWidth = 800;
+    const markerIcon = renderComponent();
+    expect(markerIcon.exists('div.oo-marker-bubble')).toBeTruthy();
+    expect(markerIcon.find('div.oo-marker-bubble').prop('style')).toHaveProperty('backgroundColor', '#0000');
+  });
+
+  it('Should render a markerIcon wtih a global background color (skin.json)', function () {
```
Comment (ipadilla4) [inline tests/components/markers/markerIcon-test.js]: Little typo here, should be "with"

File: js/components/scrubberBar.jsx [hunk 3]
```diff
@@ -291,9 +294,23 @@ class ScrubberBar extends React.Component {
...
+    if (
+      event.target.className.match('oo-marker-bubble')
+      || event.target.parentElement.className.match('oo-marker-bubble')
+    ) {
+      return;
+    }
 
-    this.setState({ hoveringX: event.nativeEvent.offsetX });
+    controller.setScrubberBarHoverState(true);
+    // eslint-disable-next-line prefer-destructuring
```
Comment (ooyala-amohnatkin) [inline js/components/scrubberBar.jsx]: please don't disable this rule here, use destructuring

File: js/components/scrubberBar.jsx [hunk 4]
```diff
@@ -283,6 +285,7 @@ class ScrubberBar extends React.Component {
    */
   handleScrubberBarMouseOver = (event) => {
     const { controller, skinConfig } = this.props;
+    let offsetX;
```
Comment (ooyala-amohnatkin) [inline js/components/scrubberBar.jsx]: it seems that we don't need to declare it here

File: js/components/markers/helpers.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+const MAXSIZE = 2;
+
+/**
+ * getPosition return the position in pixels for the markers
+ *
+ * @param {*} duration duration in seconds of the asset
```
Comment (ooyala-amohnatkin) [inline js/components/markers/helpers.js]: should it be a `number`?

File: js/components/scrubberBar.jsx [hunk 6]
```diff
@@ -507,6 +572,11 @@ class ScrubberBar extends React.Component {
         onMouseLeave={this.handleScrubberBarMouseLeave}
         onMouseMove={scrubberBarMouseMove}
       >
+        {controller.state.markers && controller.state.markers.list.length > 0 && (
```
Comment (ooyala-amohnatkin) [inline js/components/scrubberBar.jsx]: `controller.state.markers` will be rendered instead of div if `controller.state.markers.list.length <= 0`. Convert it into boolean: `{!!controller.state.markers && controller.state.markers.list.length > 0 && (`

File: tests/components/markers/markerIcon-test.js [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+      }
+    },
+    level: 0
+  };
+
+  function renderComponent(props) {
+    return Enzyme.mount(<MarkerIcon {...props}/>);
+  };
+
+  it('Shouldn\'t render a markerIcon', function () {
```
Comment (ooyala-amohnatkin) [inline tests/components/markers/markerIcon-test.js]: please add a more verbose description of why it shouldn't  be rendered

File: js/components/scrubberBar.jsx [hunk 8]
```diff
@@ -507,6 +572,11 @@ class ScrubberBar extends React.Component {
...
+          <div id="oo-marker-container">
```
Comment (ooyala-amohnatkin) [inline js/components/scrubberBar.jsx]: Is it safe to use id here? What if there are 2 players on the page?

**Comment by ooyala-amohnatkin** [comment]
@marcosmx great job!

File: js/components/scrubberBar.jsx (hunk 8 above)
Comment (marcosmx) [reply js/components/scrubberBar.jsx]: nice catch!

File: js/components/markers/helpers.js (hunk 5 above)
Comment (marcosmx) [reply js/components/markers/helpers.js]: Hahaha I just forget to add the type here.

File: scss/responsive/_xsmall.scss [hunk 9]
```diff
@@ -427,6 +427,9 @@
         }
       }
     }
+    #oo-marker-container {
```
Comment (ooyala-amohnatkin) [inline scss/responsive/_xsmall.scss]: this should be a class too

Comment (marcosmx) [reply scss/responsive/_xsmall.scss]: done, thanks for notice that!!!

File: js/components/markers/markerIcon.jsx [hunk 10]
```diff
@@ -95,21 +95,22 @@ class MarkerIcon extends Component {
...
       case CONSTANTS.MARKERS.TYPE.TEXT: {
-        if (!data.text || data.text.length === 0) {
+        if (!marker.text || marker.text.length === 0) {
           return null;
         }
-        const truncatedText = data.text.length <= CONSTANTS.MARKERS.MAXCHAR
-          ? data.text : data.text.slice(0, CONSTANTS.MARKERS.MAXCHAR).concat(' ...');
+        const truncatedText = marker.text.length <= CONSTANTS.MARKERS.MAXCHAR
+          ? marker.text : marker.text.slice(0, CONSTANTS.MARKERS.MAXCHAR).concat(' ...');
         content = <p>{truncatedText}</p>;
```
Comment (ooyala-amohnatkin) [inline js/components/markers/markerIcon.jsx]: Isn't it better to return immediately as you do on lines 105 and 114 and avoid using an additional variable?

Comment (marcosmx) [reply js/components/markers/markerIcon.jsx]: I can do it, but I will need to disable the following rule on the ESLINT config:
https://eslint.org/docs/rules/no-unreachable

Comment (marcosmx) [reply js/components/markers/markerIcon.jsx]: Nervermind, you are right....since I'm not doing anything else I can avoid the usage of the `break` statement and simply return the desired content...thank you for the observation!!!
